scripts/covid_hosp_prod.R

```diff
@@ -11,15 +12,28 @@ truth_data_date <- "2023-09-01"
 # today, which is a Wednesday. Sometimes, if we're doing a delayed forecast,
 # it's a Thursday. It's used for stamping the data and for determining the
 # appropriate as_of when creating the forecast.
-forecast_generation_date <- Sys.Date()
+forecast_generation_dates <- Sys.Date()
 # Usually, the forecast_date is the same as the generation date, but you can
 # override this. It should be a Wednesday.
-forecast_date <- round_date(forecast_generation_date, "weeks", week_start = 3)
+forecast_dates <- round_date(forecast_generation_dates, "weeks", week_start = 3)
 # forecast_generation_date needs to follow suit, but it's more complicated
 # because sometimes we forecast on Thursday.
-# forecast_generation_date <- c(as.Date(c("2024-11-20", "2024-11-27", "2024-12-04", "2024-12-11", "2024-12-18", "2024-12-26", "2025-01-02")), seq.Date(as.Date("2025-01-08"), Sys.Date(), by = 7L))
+# forecast_generation_dates <- c(as.Date(c("2024-11-20", "2024-11-27", "2024-12-04", "2024-12-11", "2024-12-18", "2024-12-26", "2025-01-02")), seq.Date(as.Date("2025-01-08"), Sys.Date(), by = 7L))
 # If doing backfill, you can set the forecast_date to a sequence of dates.
-# forecast_date <- seq.Date(as.Date("2024-11-20"), Sys.Date(), by = 7L)
+# forecast_dates <- seq.Date(as.Date("2024-11-20"), Sys.Date(), by = 7L)
+
+# Whether we're running in backtest mode.
+# If TRUE, we don't run the report notebook, which is (a) slow and (b) should be
+# preserved as an ASOF snapshot of our production results for that week.
```

Comment:
So since we embed the generation date in the notebook, it is actually preserved as a snapshot regardless. Then again, given that we're introducing scoring notebooks with this PR, it may not matter too much to retroactively generate report notebooks

Reply:
But our code is not snapshotted and during the season we made a bunch of tweaks.